Elf

calebeden opened the pull request:
Adds ELF parsing crate
Should have everything we need for LOAD sections of statically linked binaries (like 439), plus a bit more.

Incomplete Features: (some can/should maybe be done outside the library by the loader/linker?)
- Big Endian parsing (ARM can theoretically do either LE or BE, not sure if it's worth the investment)
- The relations between the different components and the string tables
- Corner cases relating to xindex
- Parsing for dynamic program header contents

ameyer1024 commented:
The `.o` and `.d` files for the example binaries should be removed and added to a `.gitignore` in the elf directory; also, could you include the script/Makefile used to generate them?

Another thing to fix is to rebase this on the main branch, to keep it as a linear history.  (Squashing it into one commit also works, but it'd be nice to preserve the commits here, since they're organized and logically distinct.)  The only conflict should be in the workspace `Cargo.toml` (just try to keep it alphabetically ordered).

calebeden commented:
@ameyer1024 I fixed the linter issues as discussed this afternoon and fixed the directory structure. I also deleted the .d files (I wasn't actually generating them from a makefile, I was just copying makefile commands from 439 and didn't delete the -MD). However, I was planning on keeping the .o files because they are useful for testing relocation sections.

ameyer1024 reviewed (approved):
Looks good; merging.
